tools/defaults.bzl

```diff
@@ -258,3 +259,36 @@ def ng_web_test_suite(deps = [], static_css = [], bootstrap = [], tags = [], **k
         tags = ["native"] + tags,
         **kwargs
     )
+
+def _devmode_js_sources_impl(ctx):
```

Comment:
Can we move this into a separate file? This is not a `default` so it should not be in `defaults.bzl`.

I'd propose `tools/es5-named-output.bzl` and the macro should be more specific to our case. Since we now need to copy this anyway, we should only use stuff we need (we need no manifest).

The rule should be named `es5_named_output` with a comment explaining what it does. Preferably we also link to https://github.com/bazelbuild/rules_nodejs/blob/a167311c025be2a77ba0d84e6a2ddcafe1c0564d/packages/typescript/src/internal/build_defs.bzl#L312 for context.